### DISCUSSION TITLE
Refoss migration

### DISCUSSION THREAD

**Discussion by javico88** [krahabb/meross_lan#344] [Q&A]
Hi, I'm pretty new to Home Assistant. I have finally installed HA Supervised in Linux MX. 
I bought a Refoss switch for a lamp and I added it to the ehomelife app but immediately I was asked to migrate my account to the new Refoss app. I did it and I can control my lamp with the app.
When a integrated the Meross Lan integration to HA, I tried to connect it with my Refoss account but it says that the email doesn't exist. I thought that Meross and Refoss share the endpoint but something is not working as I expected.
Could anyone help me to integrate my switch to Home Assistant?
Thank you

**Comment by krahabb** [comment]
Hello @javico88,
I don't have any clue about Refoss brand. Just read somewhere they're 'rebranded' Meross devices (or maybe licensed) and they should share the same api.
I think this was confirmed by users succesfully integrating their Refoss devices with meross_lan 'HTTP mode' i.e. you'd have to manually configure the device with its address/key in meross_lan for this to work just locally
Configuring the 'cloud profile' instead might be impossible (at least at the moment) since, like you've experienced, it looks like they're not sharing the cloud infrastucture.
Maybe the api is exactly the same but, like you're guessing, the endpoints are not (for this case it would be nice and likely easy to upgrade meross_lan to fully support it)

This thread might open up the discussion (and contributions) from people having a better knowledge of what's behind this brand.

**Reply by javico88** [reply]
Thanks a lot for the clarification. I have got the IP address of my device. Could you lead me for the correct way of getting the device key?

**Reply by punyearthling76** [reply]
My credentials go to a refoss.net within the refoss iOS app. I'm stuck as well.

**Reply by krahabb** [reply]
Having no devices myself it's hard for me to tell the correct way to have them setup in meross_lan.
I see there's also an official 'REFOSS integration' for HA so it's maybe easier to use.
As shared here and in other posts, these devices are pretty similar to Meross ones and should work in meross_lan too.

Maybe you'd have to setup them in meross_lan by manually adding their configuration by specifing ip address and key (could be empty if I'm not wrong for Refoss devices) or follow [CTJohnK](https://github.com/krahabb/meross_lan/discussions/344#discussioncomment-11712856) reply.

**Comment by CTJohnK** [comment]
I have several Refoss wall switches and smart plugs.  I used the Meross app to set them up (I also have many Meross devices).  You might want to setup a Meross account and reset/install the Refoss devices with the Meross app.

Then you shouldn't have any issues using this integration in Home Assistant.
